src/motion_input_shaping/shaping_comparison_demo.py

```diff
@@ -117,11 +117,14 @@ def plot(datasets: list[StepResponseData], labels: list[str], colors: list[str])
         try:
             num_lockstep_groups_possible = device.settings.get("lockstep.numgroups")
             for group_num in range(1, int(num_lockstep_groups_possible) + 1):
-                axis_nums = device.get_lockstep(group_num).get_axis_numbers()
-                if AXIS_INDEX in axis_nums:
-                    print(f"Axis {AXIS_INDEX} is part of Lockstep group {group_num}.")
-                    LOCKSTEP_INDEX = group_num
-                    break
+                try:
+                    axis_nums = device.get_lockstep(group_num).get_axis_numbers()
+                    if AXIS_INDEX in axis_nums:
+                        print(f"Axis {AXIS_INDEX} is part of Lockstep group {group_num}.")
+                        LOCKSTEP_INDEX = group_num
+                        break
+                except LockstepNotEnabledException:
+                    pass
```

Comment:
You can simplify the try catch block by catching more than one exception at the same level (reduce nestedness)
https://stackoverflow.com/questions/6470428/catch-multiple-exceptions-in-one-line-except-block